### ISSUE TITLE
Editor length limit

### ISSUE THREAD

**Issue by TomB2** [splitbrain/dokuwiki#3508]
I encounter a strange problem when editing a large site using dokuwiki's internal editor. Tested on Firefox and Chrome.

The site is rendered correctly. Only noticable difference is that the section editor generated by fastwiki does not show up for this site (but does for all others). Disabling fastwiki does not solve the issue.

Once the site has a certain length, the editor does not allow preview, save and cancel. The editor window stays open
In my case this happens in latest HogFather with content of exactly 13177 characters in 245 lines. 13178 characters would fail. (Please no discussions to split data).

PHP 7.4 settings are fairly default. According to other posts it might be related to them. I've changed them as suggested without success:
pcre.backtrack_limit=20971520
memory_limit = 600M
max_input_time = 230
max_execution_time = 240

PHP error log shows
[27-May-2021 14:40:37 Asia/Kuwait] PHP Notice: Unknown: file created in the system's temporary directory in Unknown on line 0
[27-May-2021 14:40:37 Asia/Kuwait] PHP Warning: Unknown: Unable to create temporary file, Check permissions in temporary files directory. in Unknown on line 0
[27-May-2021 14:40:37 Asia/Kuwait] PHP Warning: Unknown: POST data can't be buffered; all data discarded in Unknown on line 0

As all other pages work the I treat the error text as misleading.

Only workaround is to directly edit the file on the webserver in ...\data\pages... directory.

The problem occurs with Hogfather. Earlier versions do not seem to be affected

**Comment by splitbrain** [comment]
> Unable to create temporary file, Check permissions in temporary files directory

Did you check those permissions? Seems like PHP is trying to create a temporary file and fails.

**Comment by TomB2** [comment]
Fixed it. It was actually coincidentally with the update to HogFather
The default php.ini settings as described here "https://forum.dokuwiki.org/d/16820-header-information-post-data-temporary-file-error-message-long-page" are large enough for my wiki's purposes.

For me the reason was that the temp directory for php (C:\Windows\Temp) did not have the correct acces rights. IIS_USRS does not seem to play a role so I granted Full control for "Users" which works.

Reason to report that as bug was that editing of smaller wiki pages still worked.
